3rd-party/wpml.php

```diff
@@ -32,4 +32,51 @@ function grunion_contact_form_field_html_filter( $r, $field_label, $id ){
 	return $r;
 }
 
+/**
+ * Add current language as suffix to transient name.
+ *
+ * @param string $transient Transient name. Expected to not be SQL-escaped.
+ * @return string Translated transient name.
+ */
+add_filter( 'jetpack_sitemap_transient', 'jetpack_wpml_transient_language', 10, 1 );
+add_filter( 'jetpack_news_sitemap_transient', 'jetpack_wpml_transient_language', 10, 1 );
+add_filter( 'jetpack_sitemap_xsl_transient', 'jetpack_wpml_transient_language', 10, 1 ) ;
+
+function jetpack_wpml_transient_language( $transient ) {
+    return $transient . '_' . apply_filters( 'wpml_current_language', NULL );
+}
+
+/**
+ * Clear the multilingual sitemap cache when a sitemap action has changed.
+ *
+ * @param string $transient Transient name. Expected to not be SQL-escaped.
+ */
+add_action( 'delete_transient_jetpack_sitemap', 'jetpack_wpml_transient_delete', 10, 1 ) ;
+add_action( 'delete_transient_jetpack_news_sitemap', 'jetpack_wpml_transient_delete', 10, 1 );
+
+function jetpack_wpml_transient_delete( $transient ) {
```

Comment:
Also here, the inline doc written a few lines above must be moved to this location.